gridappsd/simulation.py

```diff
@@ -1,63 +1,140 @@
+from __future__ import absolute_import, print_function
+
+from copy import deepcopy
 import json
-import sys
-import time
+import logging
+
+from . import topics as t
+from .topics import simulation_input_topic
 
-from . topics import simulation_input_topic, simulation_output_topic, simulation_log_topic
+_log = logging.getLogger(__name__)
 
 
 class Simulation(object):
-    def __init__(self, gapps, simulation_id: str, duration: int, timestep_finished=None):
-        self.gappsd = gapps
-        self.simulation_id = simulation_id
-        self._show_timesteps = True
-        self._duration = int(duration)
-        self._timestep_requested = 0
-        self._timestep_finished_callback = timestep_finished
-        self._is_running = True
-        self.gappsd.subscribe(simulation_log_topic(simulation_id), self._on_message)
-
-    def _on_message(self, headers, message):
-        if 'logMessage' in message:
-            log_message = message['logMessage']
-            if log_message.startswith("done with timestep"):
-                timestep = int(log_message[len("done with timestep "):])
-                if self._timestep_finished_callback is not None:
-                    self._timestep_finished_callback(self, timestep)
-                # We use -1 here because the last timestep from fncs-goss-bridge.py never actually
-                # comes through at least as of 2019-6-7
-                if timestep >= self._duration-1:
-                    self._is_running = False
-
-        if self._show_timesteps:
-            # print(headers, message)
-            # print(headers)
-            sys.stdout.write(f"{message['logMessage']}\n")
-            # print(message["logMessage"])
-
-    @property
-    def running(self):
-        return self.is_running()
-
-    @property
-    def showing_status(self):
-        return self._show_timesteps
+    """ Simulation object allows controlling simulations through a python API.
+
+    The simulation object allows controlling and monitoring of simulations through
+    a python API.  It is capable of starting, stopping, pausing and restarting simulations
+    that are run on GridAPPSD.
+
+    There are three events that can be monitored by multiple callbacks:
```

Comment:
@craig8 The docstring should specify what parameters get passed to the callbacks.